### PR TITLE
- add support for Dictionary<object, object> as pseudo luaTable

### PR DESCRIPTION
**Pull request by SimonG96**

### CODE REVIEW

File: tests/src/LuaTests.cs [hunk 1]
```diff
@@ -2517,6 +2517,25 @@ public void PassIntegerToLua()
             }
         }
 
+
+        [Test]
+        public void PassIDictionaryToLua()
```
Comment (theKBro) [inline tests/src/LuaTests.cs]: It's not an IDictionary

File: tests/src/LuaTests.cs [hunk 2]
```diff
@@ -2517,6 +2517,25 @@ public void PassIntegerToLua()
...
+        {
+            //Dictionary<object, object> dictionary = new Dictionary<object, object>() {{"test", "testValue"}};
```
Comment (theKBro) [inline tests/src/LuaTests.cs]: remove comments

File: tests/src/TestTypes/TestClass.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
 ﻿using System;
+using System.Collections;
+using System.Collections.Generic;
```
Comment (theKBro) [inline tests/src/TestTypes/TestClass.cs:2]: still needed`?

**Review by theKBro — CHANGES_REQUESTED** [review]

File: tests/src/LuaTests.cs (hunk 1 above)
Comment (SimonG96) [reply tests/src/LuaTests.cs]: done

File: tests/src/LuaTests.cs (hunk 2 above)
Comment (SimonG96) [reply tests/src/LuaTests.cs]: done

File: tests/src/TestTypes/TestClass.cs (hunk 3 above)
Comment (SimonG96) [reply tests/src/TestTypes/TestClass.cs:2]: done

**Review by theKBro — APPROVED** [review]

**Pull request merged**